sdk/src/main/java/io/radar/sdk/model/RadarCoordinate.kt

```diff
@@ -11,6 +11,34 @@ class RadarCoordinate(
     val longitude: Double
 ) {
 
+    internal companion object {
+        private const val FIELD_COORDINATES = "coordinates"
+
+        @JvmStatic
+        fun fromJson(obj: JSONObject?): RadarCoordinate? {
+            if (obj == null) {
+                return null
+            }
+
+            val coordinatesObj = obj.optJSONArray(FIELD_COORDINATES)
+            val longitude = coordinatesObj?.optDouble(0) ?: 0.0
+            val latitude = coordinatesObj?.optDouble(1) ?: 0.0
+
+            return RadarCoordinate(latitude, longitude)
+        }
+
+        @JvmStatic
+        fun fromJson(arr: JSONArray?): Array<RadarCoordinate>? {
+            if (arr == null) {
+                return null
+            }
+
+            return Array(arr.length()) { index ->
+                fromJson(arr.optJSONObject(index))
+            }.filterNotNull().toTypedArray()
+        }
```

Comment:
iOS returns nil when any array element's deserialization call returns nil. 

Here we filter out that array element,